pkg/compactor/compactor.go

```diff
@@ -403,21 +433,25 @@ func (c *Compactor) compactUsers(ctx context.Context) error {
 
 		// Ensure the user ID belongs to our shard.
 		if owned, err := c.ownUser(userID); err != nil {
+			c.compactionRunSkippedTenants.Inc()
```

Comment:
Or `compactionRunFailedTenants`?

Reply:
I think skipped is more correct in this case because we're actually skipping that tenant. The `compactionRunFailedTenants` is when a tenant is not skipped and its compaction has failed.

Reply:
We are skipping tenant, but not because tenant **doesn't belong** to the compactor, but because of **failure to check** if tenant belongs to the compactor. It's not a failed compaction, so I agree that `compactionRunFailedTenants` is not the best metric. Ring has a special metric for this, perhaps compactor should have one as well?

Reply:
Not sure how much it's worth adding another metric, considering the actual reasons why it can happen. I believe another metric could just add more confusion to the user, especially in this context where we want to watch the progress of a compaction run.

Reply:
Merging to move forward, but we can continue the discussion and I will work on a follow-up PR if we decide to act differently.